diff -u shim-15/Make.defaults /home/bradley/src/shim-15/Make.defaults
--- shim-15/Make.defaults	2018-04-06 04:49:17.000000000 +1000
+++ /home/bradley/src/shim-15/Make.defaults	2019-09-23 11:15:51.404996072 +1000
@@ -68,7 +68,7 @@
 		   $(CLANG_BUGS) -m64 \
 		   -DEFI_FUNCTION_WRAPPER -DGNU_EFI_USE_MS_ABI \
 		   -DNO_BUILTIN_VA_FUNCS -DMDE_CPU_X64 -DPAGE_SIZE=4096
-	LIBDIR			?= $(prefix)/lib64
+	LIBDIR			?= $(prefix)/lib
 	ARCH_SUFFIX		?= x64
 	ARCH_SUFFIX_UPPER	?= X64
 	ARCH_LDFLAGS		?=
@@ -77,7 +77,7 @@
 	CFLAGS	+= -mno-mmx -mno-sse -mno-red-zone -nostdinc \
 		   $(CLANG_BUGS) -m32 \
 		   -DMDE_CPU_IA32 -DPAGE_SIZE=4096
-	LIBDIR			?= $(prefix)/lib
+	LIBDIR			?= $(prefix)/lib32
 	ARCH_SUFFIX		?= ia32
 	ARCH_SUFFIX_UPPER	?= IA32
 	ARCH_LDFLAGS		?=
